extensions/annotations/src/main/java/io/opentelemetry/extension/annotations/WithSpanAttribute.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation marks that a parameter of a method annotated by the {@link WithSpan} annotation
+ * should be added as an attribute to the newly created {@link io.opentelemetry.api.trace.Span}.
+ *
+ * <p>Application developers can use this annotation to signal OpenTelemetry auto-instrumentation
+ * that a new span should be created whenever marked method is executed.
+ *
+ * <p>If you are a library developer, then probably you should NOT use this annotation, because it
+ * is non-functional without the OpenTelemetry auto-instrumentation agent, or some other annotation
+ * processor.
+ *
+ * @see <a href="https://github.com/open-telemetry/opentelemetry-auto-instr-java">OpenTelemetry
+ *     Auto-Instrumentation</a>
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface WithSpanAttribute {
+  /**
+   * Optional name of the attribute.
+   *
+   * <p>If not specified, an appropriate default name should be created by auto-instrumentation.
```

Comment:
Since this can be used on multiple parameters, wouldn't that be more `classname.method.argname`? Or should this be a required parameter?

Reply:
oops, that is total copypasta on my part.  I would think that the parameter name itself would be sufficient, although note the caveat mentioned in above that this would require that the code be compiled with `-parameters` which should definitely be mentioned in this javadoc comment.

Reply:
We should phrase our javadoc for the end user, not instrumentation authors, in general I think (if notes for instrumentation are appropriate, I'd rather have them in a .md file than javadoc). So this would be something like

```
If not specified and the code is compiled using the `{@code -parameters}` argument to `javac`, the parameter name will be used instead. If the parameter name is not available, e.g., because the code was not compiled with that flag, the attribute will be ignored.
```